Cringe/Services/PpService.cs

```diff
@@ -1,18 +1,35 @@
 ﻿using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using Cringe.Types.Database;
 using Microsoft.Extensions.Logging;
-using OppaiSharp;
-using Beatmap = OppaiSharp.Beatmap;
+using osu.Framework.Audio.Track;
+using osu.Framework.Graphics.Textures;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.Formats;
+using osu.Game.Beatmaps.Legacy;
+using osu.Game.IO;
+using osu.Game.Rulesets.Osu;
+using osu.Game.Rulesets.Osu.Difficulty;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Skinning;
 
 namespace Cringe.Services
 {
     public class PpService
     {
         private readonly BeatmapService _beatmapService;
+
         private readonly ILogger<PpService> _logger;
 
+        //other game modes fuck you
+        private static readonly OsuRuleset _ruleset = new();
```

Comment:
Rulesets are meant to be created in place, they're very lightweight. I don't think theres any reason to store it here